### PR TITLE
MXWAR-145: test: add unit tests for date-utils (44 tests, 95.65% coverage)

### PR DESCRIPTION
**Pull request by Sahilbasu5101**
## Jira Ticket
https://mifosforge.jira.com/browse/MXWAR-145

## Description
Adds a Vitest unit test suite for `src/lib/date-utils.ts`.

**Changes:**
- Install `vitest` and `@vitest/coverage-v8` as devDependencies
- Configure Vitest in `vite.config.ts` (using `vitest/config` defineConfig)
- Add `npm test`, `test:watch`, `test:coverage` scripts
- 44 tests covering `formatDate()`, `dateArrayToInputValue()`, `inputToFineractDate()`

**Coverage:**
- Lines: 95.65%
- Branches: 87.87%

## -> Design Question for Maintainers
`inputToFineractDate("2023-02-29")` returns `"29 March 2023"` — month rolls
over via JS Date but original day (29) is kept. Should this return `undefined` instead?

## Summary by CodeRabbit

- **Tests**
  - Added automated coverage for date formatting and conversion utilities, including invalid inputs, leap years, locale handling, date arrays, and fallback values.
  - Added commands for running tests, monitoring changes, and generating coverage reports.
  - Expanded validation for malformed and edge-case date values.

- **Chores**
  - Configured the project’s test runner and coverage reporting for library utilities.
  - Set up a Node-based test environment with coverage focused on library code.

### CODE REVIEW

**Comment by Sahilbasu5101** [comment]
Hi @staru09 , just a gentle follow-up on this PR.
All checks are passing, and the PR is currently waiting for review.
Whenever you get some time, could you please take a look at it?
I’d really appreciate your feedback. Thank you!

**Comment by Sahilbasu5101** [comment]
hi @Craig-Rosario , have a look on this pr when ever you got some time.

**Review by IOhacker — DISMISSED** [review]

**Comment by IOhacker** [comment]
@Sahilbasu5101 please help me to resolve the conflicts

**Comment by Sahilbasu5101** [comment]
Hi @IOhacker , I have resolved the merge conflicts and rebased the branch on latest dev. All checks and unit tests are passing. It is ready to be merged. Thanks!

File: src/lib/__tests__/date-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+    it('returns fallback for whitespace-only string', () => {
+      expect(formatDate('   ')).toBe('—')
+    })
+  })
+
+  // ── ISO string inputs ───────────────────────────────────────────────────
+  describe('ISO string input', () => {
+    it('formats a valid ISO date string in en-US locale', () => {
+      const result = formatDate('2024-01-15', 'en-US')
```
Comment (devvaansh) [inline src/lib/__tests__/date-utils.test.ts]: Ran this locally on a3f2584 & this still uses an ISO string, so it fails on any negative UTC offset (January 14, 2024 in America/Los_Angeles). Same fix as the neighbours.

**Review by devvaansh — CHANGES_REQUESTED** [review]

File: src/lib/__tests__/date-utils.test.ts (hunk 1 above)
Comment (devvaansh) [reply src/lib/__tests__/date-utils.test.ts]: To be precise about "same fix",  I'd use `formatDate('2024-01-15T00:00:00', 'en-US')` rather than an array here. An explicit time parses as local instead of UTC, and it keeps this block actually testing a string; after the earlier fixes it's the only valid-ISO-string case left.


      const result = formatDate('2024-01-15T00:00:00', 'en-US')

**Comment by Sahilbasu5101** [comment]
Thanks @devvaansh , 
Updated to formatDate('2024-01-15T00:00:00', 'en-US') so it parses as local time across all timezones while keeping it an ISO string test. Verified locally across negative UTC offsets as well. Pushed the update

**Review by devvaansh — APPROVED** [review]
lgtm

**Comment by Sahilbasu5101** [comment]
@IOhacker
